### PR TITLE
[v2] Small wizard refactorings

### PR DESCRIPTION
**Pull request by kyleknap**
Specifically includes:
* Create explicit class for the `Application` and `Layout` classes. This better defines the public interfaces for those objects.
* Override the built-in exception handler to just exit the application with the captured error on any unhandled errors.

### CODE REVIEW

File: awscli/customizations/wizard/app.py [hunk 1]
```diff
@@ -39,6 +63,40 @@ def run(self, loaded):
...
+        self.error_bar_visible = None
+        super().__init__(
+            layout=layout, style=style, key_bindings=key_bindings,
+            full_screen=full_screen
+        )
+
+    def run(self, pre_run=None, **kwargs):
+        loop = get_event_loop()
+        previous_exc_handler = loop.get_exception_handler()
+        loop.set_exception_handler(self._handle_exception)
```
Comment (vz10) [inline awscli/customizations/wizard/app.py:86]: Probably makes sense to add test to check that we exit on exception.

**Review by vz10 — CHANGES_REQUESTED** [review]
I really like this refactoring, just one small comment

**Comment by kyleknap** [comment]
Ok finally got a test case working for the exception handler. Should be good to look at now.

**Review by vz10 — APPROVED** [review]
Thank you. 🚀

**Pull request merged**